### PR TITLE
Solar body augment

### PR DESCRIPTION
**Pull request by Fueredoriku**
A solar body augment that fills gun magazine when players face the correct angle
- Added reloading mechanic

More polish is wanted, but will be postponed for later

Updating the weapon transform in fps mode is going to be separate PR.

### CODE REVIEW

File: Assets/Scripts/Augment/AugmentImplementations/Hat/HatBarrelModel.cs [hunk 1]
```diff
@@ -19,42 +19,36 @@ public class HatBarrelModel : MonoBehaviour
...
-    void Start()
+    public void OnInitialize(int magazine)
     {
-        remainingBullets = magazineSize;
+        magazineSize = magazine;
         for (int i = 0; i < magazineSize; i++)
         {
             ammunition.Add(Instantiate(hatPrefab, ammunitionHolder.transform));
             ammunition[i].transform.Translate(new Vector3(0f, bulletHeigth * i, 0f));
         }
```
Comment (toberge) [inline Assets/Scripts/Augment/AugmentImplementations/Hat/HatBarrelModel.cs:30]: This whole class feels like something that could work for most barrels. Perhaps we could refactor it (later)?

File: Assets/Scripts/Augment/GunBody.cs [hunk 2]
```diff
@@ -12,4 +12,33 @@ public class GunBody : MonoBehaviour
 
     // Where to attach barrel
     public Transform attachmentSite;
+
+    [SerializeField, Range(0,1)]
+    protected float reloadEfficiencyPercentage = 1f;
+
+    //TODO: Modifier refactor
+    protected GunController gunController;
```
Comment (toberge) [inline Assets/Scripts/Augment/GunBody.cs:20]: Elaborate? I assume you mean that we should give access to the gun controller in some other way?

File: Assets/Scripts/Augment/GunController.cs [hunk 3]
```diff
@@ -36,8 +37,23 @@ private void FixedUpdate()
         }
     }
 
+    /// <summary>
+    /// Expects a percentage of ammunition to be reloaded.
+    /// This percentage is normalized eg. min = 0, max = 1.
```
Comment (toberge) [inline Assets/Scripts/Augment/GunController.cs:42]: Ummmm so it's just a fraction. No need to call it a percentage when it isn't one.

**Review by toberge — CHANGES_REQUESTED** [review]
Add solar body to auction stage and staticinfo, please...

Otherwise just nitpicks

File: Assets/Scripts/Augment/AugmentImplementations/Hat/HatBarrelModel.cs (hunk 1 above)
Comment (Fueredoriku) [reply Assets/Scripts/Augment/AugmentImplementations/Hat/HatBarrelModel.cs:30]: Yes, i would very much like to so!

I'll do it when i refator attachmentPoints and outputs

**Review by toberge — APPROVED** [review]
Sure I suppose it is fine now.

File: Assets/Scripts/Augment/GunBody.cs (hunk 2 above)
Comment (Fueredoriku) [reply Assets/Scripts/Augment/GunBody.cs:20]: There is bad coupling between gunController and augments. I'll refactor that in another PR

File: Assets/Scripts/Augment/GunController.cs (hunk 3 above)
Comment (Fueredoriku) [reply Assets/Scripts/Augment/GunController.cs:42]: Ye, good point

**Pull request merged**